tests/Unit/Adapter/Image/ImagePathFactoryTest.php

```diff
@@ -0,0 +1,89 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace Tests\Unit\Adapter\Image;
+
+use Generator;
+use PHPUnit\Framework\Assert;
+use PHPUnit\Framework\TestCase;
+use PrestaShop\PrestaShop\Adapter\Image\ImagePathFactory;
+
+class ImagePathFactoryTest extends TestCase
+{
+    /**
+     * @dataProvider getArgumentsForSmokeTest
+     *
+     * @param string $pathToBaseDir
+     */
+    public function testConstructImagePathFactory(string $pathToBaseDir): void
+    {
+        $imagePathFactory = new ImagePathFactory($pathToBaseDir);
+        Assert::assertInstanceOf(ImagePathFactory::class, $imagePathFactory);
```

Comment:
```suggestion
        $this->assertInstanceOf(ImagePathFactory::class, $imagePathFactory);
```

Using static is not necessary in PhpUnit context